### PR TITLE
improved performance of log function by 10%. Added 'use strict';

### PR DESCRIPTION
**Pull request by visla**

### CODE REVIEW

File: lib/winston-logstash-udp.js [hunk 1]
```diff
@@ -70,11 +71,14 @@ LogstashUDP.prototype.log = function(level, msg, meta, callback) {
...
 
-    self.sendLog(logEntry, function() {
-        self.emit('logged', true);
-        callback(null, true);
-    });
+    // There is no need to wait for a callback as there is no guarantee that
+    // message will ever reach destination and we never get the feedback with UDP.
+    // Having self.emit('logged', true) as part of callback in sendLog would
+    // cause it to take 10% more time.
+    self.sendLog(logEntry);
```
Comment (cthayer) [inline lib/winston-logstash-udp.js]: The callback determines if the UDP message has been sent (not whether it is received).

From the nodejs docs for `dgram.Socket.send()`:

> An optional callback may be specified to detect DNS errors or for determining when it's safe to reuse the buf object. Note that DNS lookups delay the time to send for at least one tick. The only way to know for sure that the datagram has been sent is by using a callback.

I think it is worth knowing if the system has sent the UDP message.  If it doesn't, then we know that the message will definitely not be received.

However, perhaps the callback should be optional for cases where the sent status of the UDP message is not needed/desired.

Comment (visla) [inline lib/winston-logstash-udp.js]: I Agree to having this optional. maybe even turned on to check for successful sending by default.

Comment (cthayer) [inline lib/winston-logstash-udp.js]: I agree with the default being to check for successful sending.

Can you modify the pull request to reflect this change?

**Comment by cthayer** [comment]
Closing this as it is over a year old now

**Pull request closed without merging**